webapp-django/crashstats/crashstats/views.py

```diff
@@ -524,21 +498,36 @@ def home(request, product, versions=None,
     context = default_context or {}
     contains_builds = False
     product = context['product']
+    api = models.ProductVersions()
+
+    context['product'] = product
 
-    if versions is None:
-        versions = []
-        for release in context['releases'][product]:
-            if release['featured']:
-                versions.append(release['version'])
-        contains_builds = has_builds(product, versions)
+    if versions:
+        # from the URL
+        context['versions'] = [versions]
+        # only check if it has builds if a particular version was selected
+        contains_builds = bool(
+            api.get(
+                product=product,
+                version=versions,
+                is_featured=True,
+                active=True,
+                has_builds=True
+            )['total']
+        )
     else:
-        versions = versions.split(';')
-        contains_builds = has_builds(product, versions)
+        hits = api.get(
+            is_featured=True,
+            product=product,
+            active=True,
+        )['hits']
+        context['versions'] = [x['version'] for x in hits]
 
-    context['versions'] = versions
-    context['product'] = product
-    if len(versions) == 1:
-        context['version'] = versions[0]
+        # Kinda strange, but legacy inherited code.
+        # If you look at the homepage without specifying a specific
+        # version, it will always assume that the product does not
+        # contain builds.
+        contains_builds = False
```

Comment:
What my code does is the exact same thing as on existing prod. Just doing it differently with `ProductVersions`.